vlog4j-client/src/test/java/org/vlog4j/client/picocli/SaveModelTest.java

```diff
@@ -26,13 +33,57 @@
 import org.vlog4j.client.picocli.SaveModel;
 
 public class SaveModelTest {
+	private final String outputConfigurationBase = "  --save-model: %b\n  --output-model-directory: %s\n";
+	private final String dir = "directory";
+	private final String tempDir = "tempDir";
+	private final String defaultDir = "model";
+
+	@Test
+	public void validate_saveModelTrueDefaultDir_valid() throws ConfigurationException {
+		SaveModel sm = new SaveModel();
+		sm.setSaveModel(true);
+		sm.setOutputModelDirectory(defaultDir);
+		sm.validate();
+	}
 
 	@Test
 	public void validate_saveModelTrueValidDir_valid() throws ConfigurationException {
-		//default configuration
+		// default configuration
 		SaveModel sm = new SaveModel();
 		sm.setSaveModel(false);
-		sm.setOutputModelDirectory("query-results");
+		sm.setOutputModelDirectory(dir);
+		sm.validate();
+	}
+
+	@Test(expected = ConfigurationException.class)
+	public void validate_saveModelTrueEmptyDir_notValid() throws ConfigurationException {
+		SaveModel sm = new SaveModel();
+		sm.setSaveModel(true);
+		sm.setOutputModelDirectory("");
+		sm.validate();
+	}
+
+	@Test(expected = ConfigurationException.class)
+	public void validate_saveModelTrueNullDir_notValid() throws ConfigurationException {
+		SaveModel sm = new SaveModel();
+		sm.setSaveModel(true);
+		sm.setOutputModelDirectory(null);
+		sm.validate();
+	}
+
+	@Test
+	public void validate_saveModelFalseDefaultDir_valid() throws ConfigurationException {
+		SaveModel sm = new SaveModel();
+		sm.setSaveModel(false);
+		sm.setOutputModelDirectory(defaultDir);
+		sm.validate();
+	}
+
+	@Test
+	public void validate_saveModelFalseValidDir_valid() throws ConfigurationException {
+		SaveModel sm = new SaveModel();
+		sm.setSaveModel(false);
+		sm.setOutputModelDirectory(dir);
 		sm.validate();
```

Comment:
same as above (nothing tested)

Reply:
Perhaps Mockito can also be used to test that validate() does not throw any exception.